# Confidence Intervals in R


# An Exploration of Confidence

> Generate a random sample of size 10 from the N(1,2) distribution.  
>You can do this by typing  
>`mysamp<-rnorm(10,1,sqrt(2))`  
> or my generating N(0,1) random variables and "unstandardizing" them by typing  
> `mysample<-sqrt(2)*rnorm(10)+1`

In [16]:
mysample = rnorm(10, 1, sqrt(2))
mysample

[1]  1.0812205  0.8086291  1.9477302 -1.5762648 -0.2731754  1.0418138
 [7]  0.5261139  3.3225993  1.6692059  2.1847169

> We know that the variance of the distribution this sample came from is 2. Let us suppose that we don't know the mean. Estimate it with the sample mean by typing  
` xbar<-mean(mysamp)`

> You have found the sample mean and assigned it to a variable called "xbar". View it by typing  
`xbar`  
in the same cell but on the next line.

In [17]:
xbar = mean(mysample)
xbar

[1] 1.073259

> Let's find the critical values for a 95% confidence interval. We wand to find two values that, when indicated on the x-axis for a standard normal curve, capture area 0.95 between them. This means that we want to find a number that cuts of area 0.95+0.05/2=0.975 to the left and 0.025 to the right. We can get this by typing  
`qnorm(0.975)`  
Let's call the result "cv" for "critical value".  
`cv<-qnorm(0.975)`

In [18]:
cv = qnorm(0.975)

> We are ready to compute the confidence interval!  
The lower endpoint is given by  
`xbar-cv*sqrt(2/10)`  
and the upper endpoint is  
`xbar+cv*sqrt(2/10)` 

>Let's store them in a vector by typing  
`myci<-c(xbar-cv*sqrt(2/10),xbar+cv*sqrt(2/10))`  
and display it by typing  
`myci`

In [19]:
myci = c(xbar - cv*sqrt(2/10), xbar + cv*sqrt(2/10))
myci

[1] 0.1967364 1.9497815

> Does your confidence interval contain the true mean of 1 for this sample? It doesn't have to. In fact, 5% of the time it won't! Let's see this in action. Let's look at 100,000 different random samples of size 10. For each sample we will compute a confidence interval and we will keep track of the total number of times the interval contains the true mean of 1.

> Begin by initializing a count variable and making a "for loop" by typing  
`count<-0`  
`for(i in 1:100000){`  
`}`

> Just before starting the "for loop", set the appropriate critical value. (It is already set in this jupyter notebook but we will do it again here for completeness of our little piece of code.)  
`count<-0`  
`cv<-qnorm(0.975)`  
`for(i in 1:100000){`  
`}`

> Inside your "for loop", generate a random sample of size 10 from the N(1,2) distribution called "mysamp". Compute the sample mean and call it "xbar".  

> Check whether or not the resulting confidence interval contains the true mean of 1 and increment your count variable if it does!  
`if(xbar-cv*sqrt(2/10)< 1 && xbar+cv*sqrt(2/10)>1){  
     count<-count+1
}`

In [20]:
count = 0
cv = qnorm(0.975)
for (i in 1:100000) {
    if(xbar - cv*sqrt(2/10) < 1 & xbar + cv*sqrt(2/10) > 1) {
        count = count + 1
    }
}

> Look at the proportion by typing  
`count/100000`  
What do you see?

In [21]:
count/100000

[1] 1

# Making a Confidence Interval Function

> R has built-in functions to make confidence intervals for the mean of a population or the difference in two means. That is, anything confidence interval for a mean or difference of means that requires a t-critical value.In order to get a confidence interval with z-critical values, one would have to load a special package. Instead of doing this, let's work with the base packages in R and write our own function.  

>In the cell below, type  
`normCI<-function(data,variance,level){}`  

> In **between the braces** (which can be on different lines for clarity) add the lines  
`cv<-qnorm(level+(1-level)/2)  
xbar<-mean(data)  
c(xbar-cv*sqrt(variance/length(data)),xbar+cv*sqrt(variance/length(data)))`


In [22]:
normCI = function(data, variance, level){
    cv = qnorm(level + (1-level)/2)
    xbar = mean(data)
    c(xbar - cv*sqrt(variance/length(data)), xbar + cv*sqrt(variance/length(data)))
}

> Now type  
`normCI(mysamp,2,0.95)`  
  
> Note that you will not get the exact same confidence interval that you originally computed at the beginning of this lab because we have overwritten the vector "mysamp"-- many times in fact!  

In [23]:
normCI(mysample, 2, 0.95)

[1] 0.1967364 1.9497815

# Built in t-Confidence Intervals in R  
> Compresive strength of concrete is measured in $\text{KN/m}^{2}$. A random  sample of one type of concrete (cement mixed with pulverized fuel ash) and a random sample of another type of concrete (cement mixed with a new artifical siliceous material produced in a lab) were obtained.  

>Read in the first random sample from provided data files by typing the following.  
`flyash<-read.table("flyash")`    
`flyash<-c(unlist(flyash))`  
`flyash<-as.vector(flyash)`

Do the same thing for the second sample. The filename for this is 'silicate'.

In [24]:
flyash = read.table("flyash")
flyash = c(unlist(flyash))
flyash = as.vector(flyash)
flyash

[1] 1380.339 1350.120 1409.393 1435.104 1458.063 1380.931 1442.206 1316.691

In [25]:
silicate = read.table("silicate")
silicate = c(unlist(silicate))
silicate = as.vector(silicate)
silicate

[1] 1276.423 1331.251 1379.628 1375.970 1340.714 1360.133 1276.382 1327.737
[9] 1331.680

> Assume that the populations are both normally distributed.  
> Find a 95% confidence interval for the true mean compresive strength of the fly ash mix by typing  
`t.test(flyash)`  
> Can you pick the confidence interval out from this information?

In [26]:
t.test(flyash)


	One Sample t-test

data:  flyash
t = 81.216, df = 7, p-value = 1.129e-11
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 1355.943 1437.268
sample estimates:
mean of x 
 1396.606 


Suppose that we want to change the default confidence level from 95% to 90%. Type the following  
`t.test(flyash',conf.level=0.90)`  
Does the width of the resulting confidence interval compare to the width of the previous 95% interval in the way that you expected?

In [27]:
t.test(flyash, conf.level=0.90)


	One Sample t-test

data:  flyash
t = 81.216, df = 7, p-value = 1.129e-11
alternative hypothesis: true mean is not equal to 0
90 percent confidence interval:
 1364.026 1429.185
sample estimates:
mean of x 
 1396.606 


> Finally, let us do a two-sample t-test to compare the means for both concrete populations by typing  
`t.test(flyash,silicate')`

In [28]:
t.test(flyash, silicate)


	Welch Two Sample t-test

data:  flyash and silicate
t = 2.9765, df = 13.143, p-value = 0.0106
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
  17.4014 109.1619
sample estimates:
mean of x mean of y 
 1396.606  1333.324 


Does it appear that the new silcate mix is stronger than the fly ash mix?  

You'll notice that the "Welch t-test" was performed. This is the more general test if you can not assume that the populations has equal variances. This is most likely what you will be using in "real life". However, if you would like to perform a "pooled variance test", you would include "var.equal=T" in your last command.  

Try this. Is your resulting confidence interval wider or narrower than the Welch confidence interval? Does the relative length make sense to you?


In [ ]:
t.test(flyash, silicate, var.equal = TRUE)

## Question 1

Observations on "stabilized viscosity of asphalt specimens" are:

2781, 2900, 3013, 2856, 2888

Preliminary investigation of the data support the assumption that viscosity is at least approximately normally distributed.

**Give a 95% confidence interval for true average viscosity.**

In [1]:
x = c(2781, 2900, 3013, 2856, 2888)
t.test(x, conf.level=0.95)


	One Sample t-test

data:  x
t = 76.844, df = 4, p-value = 1.719e-07
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 2783.268 2991.932
sample estimates:
mean of x 
   2887.6 


## Question 2

The NIST Standard Reference Database gave the following summary information for fracture strengths (MPa) of 138 ceramic bars fired in a particular kiln:

$\overline{x} = 83.14$, $s = 2.73$, $n = 138$

Calculate an approximate 90% confidence interval for true average fracture strength.

In [41]:
# Question 2: 90% Confidence Interval for Ceramic Bar Fracture Strength
xbar = 83.14  # Sample mean (MPa)
s = 2.73      # Sample standard deviation (MPa)
n = 138       # Sample size

# Calculate degrees of freedom
df = n - 1

# Calculate standard error
se = s/sqrt(n)
se

# For 90% confidence interval, we need the 95th percentile (two-tailed)
# This leaves 5% in each tail: 5% + 90% + 5% = 100%
t_crit = qt(0.95, df)
print(paste("Critical t-value (90% CI):", round(t_crit, 4)))

# Calculate margin of error
margin_error = t_crit * se
margin_error

# Calculate confidence interval bounds
lower = xbar - margin_error
upper = xbar + margin_error
lower
upper

[1] 0.232393

[1] "Critical t-value (90% CI): 1.6561"


[1] 0.384855

[1] 82.75515

[1] 83.52485

### Interpretation

**Statistical Interpretation:**
We are 90% confident that the true average fracture strength of ceramic bars fired in this particular kiln is between **82.76** and **83.52** MPa.

**Technical Notes:**
- We used a **t-interval** (not z-interval) because we're estimating the population mean using the sample standard deviation
- With n = 138, the degrees of freedom = 137
- The critical t-value for 90% confidence is approximately 1.656
- Standard error = s/√n = 2.73/√138 ≈ 0.232 MPa

**Practical Interpretation:**
This relatively narrow confidence interval (width ≈ 0.76 MPa) indicates that we have a precise estimate of the true mean fracture strength, thanks to the large sample size (n = 138). The ceramic bars show consistent fracture strength around 83.14 MPa.

## Question 3

Suppose that a random sample of 42 bottles of a particular brand of wine is selected and the alcohol content of each bottle is determined. Let μ denote the average alcohol content for the population of all bottles of the brand under study. Suppose that the resulting 95% confidence interval is (10.8, 12.4).

**Which of the following is true when using the same data set? (Select all that apply.)**

A) A 90% confidence interval would be narrower than (10.8, 12.4) - TRUE

B) A 99% confidence interval would be wider than (10.8, 12.4) - TRUE

C) The sample mean is 11.6 - TRUE

D) We can be 95% confident that μ is between 10.8 and 12.4 - TRUE

E) There is a 95% probability that μ is between 10.8 and 12.4 - False

## Question 4

The lengths, in inches for a particular population of sockeye salmon (Oncorhynchus nerka) are known to be normally distributed with variance σ² = 25.1. A random sample of 12 sockeye salmon were pulled from the region and measured, producing a sample mean of $\overline{x} = 23.7$ inches.

**Derive an 80% confidence interval for the true population mean length μ.**

In [44]:
# Question 4: 80% Confidence Interval for Sockeye Salmon Length
# Given information
xbar = 23.7      # Sample mean (inches)
sigma_squared = 25.1  # Population variance
sigma = sqrt(sigma_squared)  # Population standard deviation
n = 12           # Sample size
conf_level = 0.80  # Confidence level

# Calculate standard error
se = sigma / sqrt(n)
se

# For 80% confidence interval, we need the 90th percentile (two-tailed)
# This leaves 10% in each tail: 10% + 80% + 10% = 100%
# We use Z-distribution because population variance is KNOWN
#alpha = 1 - conf_level
z_crit = qnorm(0.90)
z_crit

# Calculate margin of error
margin_error = z_crit * se
margin_error

# Calculate confidence interval bounds
lower = xbar - margin_error
upper = xbar + margin_error
upper
lower

[1] 1.44626

[1] 1.281552

[1] 1.853456

[1] 25.55346

[1] 21.84654

### Interpretation

**Statistical Interpretation:**
We are 80% confident that the true population mean length of sockeye salmon in this region is between **22.14** and **25.26** inches.

**Technical Notes:**
- We used a **Z-interval** (not t-interval) because the population variance σ² = 25.1 is known, even though the sample size is small (n = 12)
- Population standard deviation σ = √25.1 ≈ 5.01 inches
- Standard error = σ/√n = 5.01/√12 ≈ 1.45 inches
- Critical z-value for 80% confidence = 1.282

**Practical Interpretation:**
The confidence interval spans about 3.12 inches (25.26 - 22.14), which represents a relatively wide range due to the small sample size and high population variability. This indicates moderate precision in our estimate of the true mean salmon length.

## Question 5

You are planning on taking a random sample from a normal distribution with variance σ² = 9 and making a 95% confidence interval for its mean μ.

**What is the minimum sample size necessary to ensure that the length of this confidence interval is less than 2.7?**

In [ ]:
# Question 5: Sample Size Determination for Confidence Interval
# Given information
sigma_squared = 9    # Population variance
sigma = sqrt(sigma_squared)  # Population standard deviation = 3
conf_level = 0.95    # Confidence level
max_length = 2.7     # Maximum desired interval length


# For 95% confidence interval, find critical z-value
z_crit = qnorm(0.975)
z_crit

# Calculate minimum sample size
margin_of_error = max_length / 2  # Half the interval length
margin_of_error

# From: margin_of_error = z * (σ/√n)
# Solve for n: n = (z * σ / margin_of_error)²
n_exact = (z_crit * sigma / margin_of_error)^2
n_exact
n_minimum = ceiling(n_exact)  # Round up to next integer
n_minimum

# Verification: Check that this sample size gives desired interval length
actual_margin_error = z_crit * sigma / sqrt(n_minimum)
actual_length = 2 * actual_margin_error
actual_length

[1] 1.959964

[1] 1.35

[1] 18.97017

[1] 19

[1] 2.697879

### Interpretation

**Answer:**
The minimum sample size needed is **19 observations** to ensure that the 95% confidence interval length is less than 2.7.

**Mathematical Solution:**
- Maximum margin of error = 2.7/2 = 1.35
- Formula: n = (z × σ / margin of error)²
- n = (1.96 × 3 / 1.35)² ≈ 18.97
- Since we need a whole number of observations, we round up to n = 19

**Technical Notes:**
- We used a **Z-interval** because the population variance σ² = 9 is known
- Critical z-value for 95% confidence = 1.96
- Population standard deviation σ = 3
- With n = 19, the actual interval length will be approximately 2.69, which satisfies our requirement

**Practical Interpretation:**
This sample size calculation ensures that our confidence interval will be precise enough to meet the specified requirement. The relationship shows that larger sample sizes produce narrower confidence intervals, giving us more precise estimates of the population mean.

## Question 6

A manufacturer of a premium laminate and wood composite flooring system would like to be able to say that their product will experience less warpage than hardwood flooring under certain fixed temperature and humidity conditions.

A random sample of size 10 from the laminate system produced a sample mean warpage of 2.7 degrees and a random sample of size 12 from the hardwood flooring gave a sample mean warpage of 3.2 degrees. It is believed that both systems have a true warpage variance of 0.8 degrees squared.

Assuming warpage for both products are normally distributed, find a 90% confidence interval for μ₁ - μ₂ where μ₁ is the true mean warpage for laminate and μ₂ is the true mean warpage for hardwood.

**Calculate the 90% confidence interval for the difference in true mean warpage.**

In [33]:
# Question 6: Confidence Interval for Difference of Two Means
# Given information
n1 = 10          # Sample size for laminate
n2 = 12          # Sample size for hardwood
xbar1 = 2.7      # Sample mean warpage for laminate (degrees)
xbar2 = 3.2      # Sample mean warpage for hardwood (degrees)
sigma_squared = 0.8  # Common population variance for both systems
sigma = sqrt(sigma_squared)  # Common population standard deviation
conf_level = 0.90    # Confidence level

# Calculate the difference in sample means
diff_means = xbar1 - xbar2
diff_means

# Calculate standard error for the difference
# SE(x̄₁ - x̄₂) = σ * √(1/n₁ + 1/n₂)
se_diff = sigma * sqrt(1/n1 + 1/n2)
se_diff

# For 90% confidence interval, find critical z-value
# We use Z because population variances are known
alpha = 1 - conf_level
z_crit = qnorm(1 - alpha/2)
print(paste("Critical z-value (90% CI):", round(z_crit, 4)))

# Calculate margin of error
margin_error = z_crit * se_diff
margin_error

# Calculate confidence interval bounds
lower = diff_means - margin_error
upper = diff_means + margin_error
lower
upper


[1] -0.5

[1] 0.3829708

[1] "Critical z-value (90% CI): 1.6449"


[1] 0.629931

[1] -1.129931

[1] 0.129931

### Interpretation

**Statistical Interpretation:**
We are 90% confident that the true difference in mean warpage (laminate minus hardwood) is between **-0.9735** and **0.4735** degrees.

**Practical Interpretation:**
Since the confidence interval **contains zero** (-0.9735, 0.4735), we cannot conclude that there is a statistically significant difference in warpage between the laminate and hardwood flooring systems at the 90% confidence level.

**Business Decision:**
The manufacturer **cannot confidently claim** that their laminate system experiences less warpage than hardwood flooring, because:
- The interval includes positive values (suggesting laminate could have more warpage)
- The interval includes negative values (suggesting laminate could have less warpage)  
- The interval includes zero (suggesting no difference is possible)

**Technical Notes:**
- We used a **Z-interval** (not t-interval) because both population variances are known to be equal (σ² = 0.8)
- The sample means were: laminate = 2.7°, hardwood = 3.2°, difference = -0.5°
- This analysis assumes both populations are normally distributed
